### ISSUE TITLE
qTranslate messes up my Visual Composer

### ISSUE THREAD

**Issue by vinnygats** [qTranslate-Team/qtranslate-x#304]
Hi,

I'm happy enough with the plugin. It's the only one that fits right for my website.
Although, I noticed that, whenever I try to translate existing  pages, it messes up my Visual Composer. As a result of this, I can see my page live but I cannot see the page's modules on my Visual Composer. I tried updating the Visual Composer and I asked my theme's author too, but in vain.
I made several tests and my conclusion is that whenever I choose the alter page to edit the second language, it ruins my Visual Composer.
Is this a known issue? Can you somehow provide any help on the matter?

Best Regards, 

George

**Comment by johnclause** [comment]
How can we reproduce this? We need list of steps as described here: https://qtranslatexteam.wordpress.com/troubleshooting/. Do you also use https://wordpress.org/plugins/js-composer-qtranslate-x/?

**Comment by vinnygats** [comment]
Hi johnclause,

I've installed the Debug Bar plugin and inspected the page which I'm editing my front page with the Visual Composer. The error and warnings I receive are the following :

On the Java console I get : 
error : TypeError: this.sink is null network-monitor.js:285:5
warning : window.controllers is deprecated. Do not use it for UA detection. ace.js:1:0
error : URIError: malformed URI sequence php.default.min.js:659:35

On the Debug Bar I get : 
URIError: malformed URI sequence
http://test.pan-orama.org/wp-content/plugins/js_composer/assets/lib/php.default/php.default.min.js?ver=4.3.5 line 659
I inspected the line 659 and it's this part of code
function rawurldecode(str){return decodeURIComponent(str+'');}

And yes, I also use the WPBakery Visual Composer & qTranslate-X

Best Regards,

George

**Comment by vinnygats** [comment]
Hi, 
It looks like I solved the problem from a Wordpress forum solution.
https://wordpress.org/support/topic/visual-composer-not-showing-full-page-error-urierror-malformed-uri-sequence

The problem occurs when using the raw html element. I had to include trim here:
wp-content\plugins\js_composer\assets\js\backend\composer-atts.js

render:function (param, value) {
                return $("<div/>").text(rawurldecode(base64_decode(value))).html();
            }

Trim - >

render:function (param, value) {
                return $("<div/>").text(rawurldecode(base64_decode(trim(value)))).html();
            }

Thank you very much for the support.
